src/cmdline.rs

```diff
@@ -586,10 +686,428 @@ impl DefaultString for Architecture {
+            delete_karg: vec!["m".into(), "n".into()],
+            copy_network: true,
+            network_dir: DefaultedString::<NetworkDir>::from_str("o").unwrap(),
+            save_partlabel: vec!["p".into(), "q".into()],
+            save_partindex: vec!["r".into(), "s".into()],
+            offline: true,
+            insecure: true,
+            insecure_ignition: true,
+            stream_base_url: Some(Url::parse("http://example.com/t").unwrap()),
+            preserve_on_error: true,
+            fetch_retries: FetchRetries::from_str("3").unwrap(),
+            device: Some("u".into()),
+        };
+        let expected = vec![
+            "--stream",
+            "c",
+            "--image-url",
+            "http://example.com/d",
+            "--image-file",
+            "e",
+            "--ignition-file",
+            "f",
+            "--ignition-url",
+            "http://example.com/g",
+            "--ignition-hash",
+            "sha256-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+            "--architecture",
+            "h",
+            "--platform",
+            "i",
+            "--append-karg",
+            "k",
+            "--append-karg",
+            "l",
+            "--delete-karg",
+            "m",
+            "--delete-karg",
+            "n",
+            "--copy-network",
+            "--network-dir",
+            "o",
+            "--save-partlabel",
+            "p",
+            "--save-partlabel",
+            "q",
+            "--save-partindex",
+            "r",
+            "--save-partindex",
+            "s",
+            "--offline",
+            "--insecure",
+            "--insecure-ignition",
+            "--stream-base-url",
+            "http://example.com/t",
+            "--preserve-on-error",
+            "--fetch-retries",
+            "3",
+            "u",
+        ];
+        assert_eq!(config.to_args().unwrap(), expected);
+    }
+
+    /// Check that default InstallConfig serializes to empty arg list
+    #[test]
+    fn serialize_default_install_config() {
+        let config = InstallConfig::default();
+        let expected: Vec<String> = Vec::new();
+        assert_eq!(config.to_args().unwrap(), expected);
+    }
+
+    /// Check that minimal install config file serializes to minimal arg list
+    #[test]
+    fn serialize_empty_install_config_file() {
+        let config: InstallConfig = serde_yaml::from_str("device: foo").unwrap();
+        assert_eq!(config.to_args().unwrap(), vec!["foo"]);
+    }
+
+    /// Check that empty command line serializes to empty arg list
+    #[test]
+    fn serialize_empty_command_line() {
+        let expected = ["/dev/missing"];
+        let config = InstallConfig::from_args(&expected).unwrap();
+        assert_eq!(config.to_args().unwrap(), expected);
+    }
+
+    /// Test multiple config files overlapping with command-line arguments
+    #[test]
+    fn install_config_file_overlapping_field() {
```

Comment:
Would be good to expand this test (or add a separate test) with a YAML doc containing all the fields. Can reuse the values from `serialize_full_install_config` so we check the deserialized version against the `InstallConfig` there.